### PR TITLE
stats: garbage collect stats for partition table

### PR DESCRIPTION
**Pull request by lamxTyler**
### What problem does this PR solve? 

We need to remove the unused partition stats.

### What is changed and how it works?

Find the corresponding table schema from partition id, then remove the unused stats from the store.
### Check List 

Tests 

 - Unit test

Code changes

 - Has exported function/method change

Side effects

 - None

Related changes

 - None

PTAL @zz-jason @winoros @eurekaka

### CODE REVIEW

File: statistics/gc.go [hunk 1]
```diff
@@ -48,22 +48,24 @@ func (h *Handle) GCStats(is infoschema.InfoSchema, ddlLease time.Duration) error
...
 	// The table has already been deleted in stats and acknowledged to all tidb,
 	// we can safely remove the meta info now.
 	if len(rows) == 0 {
-		sql := fmt.Sprintf("delete from mysql.stats_meta where table_id = %d", tableID)
+		sql := fmt.Sprintf("delete from mysql.stats_meta where table_id = %d", physicalID)
 		_, _, err := h.restrictedExec.ExecRestrictedSQL(nil, sql)
 		return errors.Trace(err)
 	}
-	tbl, ok := is.TableByID(tableID)
+	h.mu.Lock()
```
Comment (winoros) [inline statistics/gc.go]: Do we need lock here? 
There'll be race in previous code?

File: statistics/gc.go [hunk 2]
```diff
@@ -48,22 +48,24 @@ func (h *Handle) GCStats(is infoschema.InfoSchema, ddlLease time.Duration) error
 	return nil
 }
 
-func (h *Handle) gcTableStats(is infoschema.InfoSchema, tableID int64) error {
-	sql := fmt.Sprintf("select is_index, hist_id from mysql.stats_histograms where table_id = %d", tableID)
+func (h *Handle) gcTableStats(is infoschema.InfoSchema, physicalID int64) error {
+	sql := fmt.Sprintf("select is_index, hist_id from mysql.stats_histograms where table_id = %d", physicalID)
```
Comment (winoros) [inline statistics/gc.go]: Do we need to change the `table_id` in the sql text?

Comment (lamxTyler) [reply statistics/gc.go]: The change needs a ddl operation and may lead to incompatibility between different version of tidb server.

**Comment by zz-jason** [comment]
@winoros @eurekaka PTAL

**Pull request merged**